build/apple_release/scripts/download-signed-mac-OS-binaries.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+cliExecutableName=$1
+releaseVersion=$2
+goarch=$3
+GITHUB_ACCESS_TOKEN=$4
+
+# This script downloads signed macOS binaries for a specific version and architecture.
+
+# Function to retrieve the specific artifact URL with retries
+get_specific_artifact_url_with_retries() {
+    local max_retries=4
+    local cooldown=15 # Cooldown in seconds between retries
```

Comment:
Comments should be above the code
```suggestion
    # Cooldown in seconds between retries
    local cooldown=15
```